src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/StreamExtensions.cs

```diff
@@ -76,5 +76,35 @@ internal static int TryReadAll(this Stream stream, byte[] buffer, int offset, in
             }
             return totalBytesRead;
         }
+
+        /// <summary>
+        /// Resolve image size as either the given user-specified size or distance from current position to end-of-stream.
+        /// Also performs the relevant argument validation and publicly visible caller has same argument names.
+        /// </summary>
+        /// <exception cref="ArgumentException">size is null and distance from current position to end-of-stream can't fit in Int32.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Size is negative or extends past the end-of-stream from current position.</exception>
+        internal static int GetAndValidateSize(Stream stream, int? size, string streamParameterName)
+        {
+            long maxSize = stream.Length - stream.Position;
+
+            if (size.HasValue)
+            {
+                if (unchecked(size.Value) > maxSize)
```

Comment:
Pre-existing move, but I think that was supposed be unchecked((uint)size.Value).